### ISSUE TITLE
fix(docker): SIGILL persists on Apple Silicon after #3127 — crash after faiss load during SearchDaemon init

### ISSUE THREAD

**Issue by windoliver** [nexi-lab/nexus#3134]
## Summary

PR #3127 fixed SimSIMD SVE compile-time flags, but `nexusd` still crashes with `Illegal instruction` (SIGILL) on Apple Silicon (M-series) inside Docker immediately after faiss loads successfully.

The latest `edge` image (built 2026-03-18T05:43Z, after #3127 merged at 03:29Z) reproduces the crash.

## Reproduction

```bash
docker pull ghcr.io/nexi-lab/nexus:edge
docker run --rm ghcr.io/nexi-lab/nexus:edge \
  nexusd --host 0.0.0.0 --port 2026 --auth-type database --profile full
```

**Environment:** Apple Silicon M-series, OrbStack (Docker), `linux/arm64` image.

## Crash log

```
[info] Vector index warmed in 15.4ms
[info] Loading faiss.
[info] Successfully loaded faiss.
/usr/local/bin/docker-entrypoint.sh: line 436: 24 Illegal instruction  nexusd --host 0.0.0.0 --port 2026 --auth-type database --profile full
```

Key observations:
- faiss **loads successfully** ("Successfully loaded faiss")
- SIGILL happens **immediately after**, during SearchDaemon initialization
- The crash is at entrypoint line 436 (vs line 419 in the pre-#3127 image), suggesting the entrypoint changed but the underlying issue remains

## Analysis

#3127 disabled SimSIMD SVE backends at compile time (`SIMSIMD_TARGET_SVE*=0`). The remaining SIGILL likely comes from one of:

1. **faiss-cpu itself** — faiss's own SVE/SVE2 runtime detection (`faiss::cpu_has_sve()`) may trigger SIGILL on Apple Silicon Docker VMs where `/proc/cpuinfo` doesn't correctly report feature flags
2. **txtai/sentence-transformers** — loaded after faiss during SearchDaemon warmup, may pull in torch ops with SVE
3. **A Rust extension** (`nexus_fast`, `nexus_raft`) — compiled with target features that include SVE instructions

## Suggested investigation

```bash
# Isolate whether it's faiss, torch, or Rust:
docker run --rm -it ghcr.io/nexi-lab/nexus:edge python3 -c "import faiss; print('faiss ok')"
docker run --rm -it ghcr.io/nexi-lab/nexus:edge python3 -c "import torch; print('torch ok')"
docker run --rm -it ghcr.io/nexi-lab/nexus:edge python3 -c "import simsimd; print('simsimd ok')"
docker run --rm -it ghcr.io/nexi-lab/nexus:edge python3 -c "import sentence_transformers; print('st ok')"
```

Setting `FAISS_OPT_LEVEL=generic` is already in the Dockerfile env but may not take effect during the import path that triggers the crash.

## Related

- #3125 — original SIGILL report
- #3127 — SimSIMD SVE fix (merged, partially addresses the issue)

**Comment by windoliver** [comment]
## Root cause found

The SIGILL is **NOT from faiss or simsimd** — it's from **`annoy`** (Spotify's Approximate Nearest Neighbors library), imported by txtai's ANN backend auto-discovery.

### Proof

```bash
# faiss, torch, sentence_transformers all import fine:
docker run --rm --entrypoint python3 ghcr.io/nexi-lab/nexus:edge -c "
import faiss; print('faiss ok')
import torch; print('torch ok')
import sentence_transformers; print('st ok')
"
# Output: faiss ok / torch ok / st ok

# But txtai crashes during import:
docker run --rm --entrypoint python3 ghcr.io/nexi-lab/nexus:edge -c "import txtai"
# Exit code 132 (SIGILL)

# Tracing shows it crashes when importing annoy:
# importing: ann → importing: annoy → importing: annoylib → SIGILL
```

### Fix

The `annoy` PyPI wheel for `linux/arm64` contains SVE instructions that crash under Docker's ARM64 virtualization on Apple Silicon. Options:

1. **Pin `annoy` to a version without SVE** or build from source with `-march=armv8-a+nosimd` 
2. **Exclude annoy from txtai install** — txtai's ANN factory has multiple backends (faiss, hnswlib, etc.), annoy is optional
3. **Set `ANNOY_COMPILER_ARGS` at build time** to disable SVE codegen
